crates/wasmparser/src/validator/component.rs

```diff
@@ -1048,6 +1086,896 @@ impl ComponentState {
         Ok(())
     }
 
+    pub fn task_backpressure(
+        &mut self,
+        types: &mut TypeAlloc,
+        offset: usize,
+        features: &WasmFeatures,
+    ) -> Result<()> {
+        if !features.component_model_async() {
+            bail!(
+                offset,
+                "`task.backpressure` requires the component model async feature"
+            )
+        }
+
+        let (_is_new, group_id) = types.intern_canonical_rec_group(
+            false,
+            RecGroup::implicit(
+                offset,
+                SubType {
+                    is_final: true,
+                    supertype_idx: None,
+                    composite_type: CompositeType {
+                        shared: false,
+                        inner: CompositeInnerType::Func(FuncType::new([ValType::I32], [])),
+                    },
+                },
+            ),
+        );
```

Comment:
You might be able to tighten this up a bit with this pattern perhaps?

https://github.com/bytecodealliance/wasm-tools/blob/00091b2aa1cbb7d700c0d16ee00eddd453f02260/crates/wasmparser/src/validator/component.rs#L1016-L1018

Reply:
Alternatively this pattern is used in a number of locations throughout this file (both added in this PR and preexisting), so if you'd like having a helper as well for this I think would also be reasonable.